app/controllers/moderations_controller.rb

```diff
@@ -42,6 +42,7 @@ def actions_panel
     has_no_relevant_adjustments = @adjustments.pluck(:tag_id).intersection(tag_mod_tag_ids).size.zero?
     can_be_adjusted = @moderatable.tags.ids.intersection(tag_mod_tag_ids).size.positive?
 
+    @is_user_flagged = Reaction.user_vomits.where(user_id: session_current_user_id, status: "confirmed").any?
```

Comment:
Hey @joaoGabriel55 👋🏾 

So in the app, I see that the `Flag user` link is showing `Unflag user` for users that I myself have not flagged.

I think the source of this behavior is where you set `@is_user_flagged`. It appears that we are not searching for the flags created by the current moderator on the _specific post author_; we are instead searching for flags created on _any authors_, which is incorrect.

I think a better search accounts for the current post author, such as:
```ruby
Reaction.user_vomits.where(user_id: session_current_user_id, reactable_id: @moderatable.user_id, status: "confirmed").any?
```

When you look at the structure of a flag reaction, the id for the flagged user is `reactable_id`:
```ruby
#<Reaction:0x00007fd16f077568
 id: 990,
 category: "vomit",
 created_at: Tue, 07 Jun 2022 20:04:06.781383000 UTC +00:00,
 points: -50.0,
 reactable_id: 5,
 reactable_type: "User",
 status: "confirmed",
 updated_at: Tue, 07 Jun 2022 20:04:06.781383000 UTC +00:00,
 user_id: 11>
```

Let me know if this makes sense. 🙏🏾